SeaPublicWebsite.BusinessLogic/Services/QuestionFlowService.cs

```diff
@@ -0,0 +1,806 @@
+                : QuestionFlowStep.GlazingType;
+        }
+
+        private QuestionFlowStep HeatingTypeBackDestination(QuestionFlowStep? entryPoint)
+        {
+            return entryPoint is QuestionFlowStep.HeatingType
+                ? QuestionFlowStep.AnswerSummary
+                : QuestionFlowStep.OutdoorSpace;
+        }
+
+        private QuestionFlowStep OtherHeatingTypeBackDestination(QuestionFlowStep? entryPoint)
+        {
+            return entryPoint is QuestionFlowStep.OtherHeatingType
+                ? QuestionFlowStep.AnswerSummary
+                : QuestionFlowStep.HeatingType;
+        }
+
+        private QuestionFlowStep HotWaterCylinderBackDestination(QuestionFlowStep? entryPoint)
+        {
+            return entryPoint is QuestionFlowStep.HotWaterCylinder
+                ? QuestionFlowStep.AnswerSummary
+                : QuestionFlowStep.HeatingType;
+        }
+
+        private QuestionFlowStep NumberOfOccupantsBackDestination(PropertyData propertyData, QuestionFlowStep? entryPoint)
+        {
+            return entryPoint is QuestionFlowStep.NumberOfOccupants
+                ? QuestionFlowStep.AnswerSummary
+                : propertyData.HeatingType switch
+                {
+                    HeatingType.Storage or HeatingType.DirectActionElectric or HeatingType.HeatPump
+                        or HeatingType.DoNotKnow
+                        => QuestionFlowStep.HeatingType,
+                    HeatingType.GasBoiler or HeatingType.OilBoiler or HeatingType.LpgBoiler
+                        => QuestionFlowStep.HotWaterCylinder,
+                    HeatingType.Other
+                        => QuestionFlowStep.OtherHeatingType,
+                    _ => throw new ArgumentOutOfRangeException()
+                };
+        }
+
+        private QuestionFlowStep HeatingPatternBackDestination(QuestionFlowStep? entryPoint)
+        {
+            return entryPoint is QuestionFlowStep.HeatingPattern
+                ? QuestionFlowStep.AnswerSummary
+                : QuestionFlowStep.NumberOfOccupants;
+        }
+
+        private QuestionFlowStep TemperatureBackDestination(QuestionFlowStep? entryPoint)
+        {
+            return entryPoint is QuestionFlowStep.Temperature
+                ? QuestionFlowStep.AnswerSummary
+                : QuestionFlowStep.HeatingPattern;
+        }
+
+        private QuestionFlowStep AnswerSummaryBackDestination()
+        {
+            return QuestionFlowStep.Temperature;
+        }
+
+        private QuestionFlowStep NoRecommendationsBackDestination()
+        {
+            return QuestionFlowStep.AnswerSummary;
+        }
+
+        private QuestionFlowStep YourRecommendationsBackDestination()
+        {
+            return QuestionFlowStep.AnswerSummary;
+        }
+
+        private QuestionFlowStep NewOrReturningUserForwardDestination()
+        {
+            // TODO: Routing for the next step of returning user?
+            return QuestionFlowStep.Country;
```

Comment:
TODO still here. Any fix for this?

Reply:
Nope, not as part of this refactoring anyway.
Actually, I'll remove this comment as some of the target pages are outside the question flow anyway so it wouldn't make sense to return then from here.